src/inline-assembly.md

```diff
@@ -37,515 +50,892 @@ assert_eq!(x, 4 * 6);
+> The stack is defined by an target-specific register and is a target-specific memory region. It may include a "red zone".
+> If the `nostack` option is *not* specified, then the stack pointer is guaranteed to point to memory that can be allocated by the asm block, which is aligned at least as much as is required by the ABI for a function call.
+
+```rust,ignore
+// The following snippet has undefined behaviour
+# #[cfg(target_arch = "x86_64")] { unsafe{
+let x: i32;
+core::arch::asm!("push 5", "pop rax", out("eax") x, options(nostack));
+# }}
+```
+
+r[asm.options.preserve_flags]
+The `preserves_flags` option may be specified. The implementation may assume that the value of the status flags are preserved by the assembly block.
+
+> [!TARGET-SPECFIC]
+> - These flags registers must be restored upon exiting the asm block if the `preserves_flags` option is set:
+> - x86
+>   - Status flags in `EFLAGS` (CF, PF, AF, ZF, SF, OF).
+>   - Floating-point status word (all).
+>   - Floating-point exception flags in `MXCSR` (PE, UE, OE, ZE, DE, IE).
+> - ARM
+>   - Condition flags in `CPSR` (N, Z, C, V)
+>   - Saturation flag in `CPSR` (Q)
+>   - Greater than or equal flags in `CPSR` (GE).
+>   - Condition flags in `FPSCR` (N, Z, C, V)
+>   - Saturation flag in `FPSCR` (QC)
+>   - Floating-point exception flags in `FPSCR` (IDC, IXC, UFC, OFC, DZC, IOC).
+> - AArch64
+>   - Condition flags (`NZCV` register).
+>   - Floating-point status (`FPSR` register).
+> - RISC-V
+>   - Floating-point exception flags in `fcsr` (`fflags`).
+>   - Vector extension state (`vtype`, `vl`, `vcsr`).
+> - LoongArch
+>   - Floating-point condition flags in `$fcc[0-7]`.
+
+```rust,ignore
+# #[cfg(target_arch = "x86_64")] { unsafe{
+core::arch::asm!("cmp eax, eax", in("eax") 5, options(preserve_flags));
+# }}
+```
+
+r[asm.options.noreturn]
+The `noreturn` option may be specifed. An invocation of the [`core::arch::asm!`] macro that specifies the `noreturn` option expands to an expression of type `!`. The behaviour is undefined if an evaluation of the assembly block exits. The program shall not specify the `clobber_abi` specification, or an operand that is an `out`, `lateout`, `inout`, or `inlateout` operand. 
+
+```rust
+# #[cfg(target_arch = "x86_64")]
+pub fn main() -> ! {
+  unsafe{
+    core::arch::asm!("xor edi, edi", "call exit@plt", options(noreturn));
+  }
+}
+```
+
+```rust,ignore
+// The following snippet has undefined behaviour
+# #[cfg(target_arch = "x86_64")] { unsafe{
+core::arch::asm!("", options(noreturn));
+# }}
+```
+
+```rust,compile_fail
+# #[cfg(target_arch = "x86_64")] { unsafe{
+let x: i32;
+core::arch::asm!("xor edi, edi", "call exit@plt", out("edi") x, options(noreturn));
+# }}
+# #[cfg(not(target_arch = "x86_64"))] compile_error!("Inline Assembly Tests are not supported off of x86_64");
+```
+
+r[asm.options.global]
+A program shall not specify an option, other than the `att_syntax` option, in an invocation of the [`core::arch::global_asm!`] macro.
```

Comment:
You can also use the `raw` option with global_asm. In fact it seems the `raw` option was omitted entirely in the spec?